test/e2e/framework/deploy/deploy.go

```diff
@@ -0,0 +1,115 @@
+// +build e2e
+
+package deploy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/exec"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/yaml"
+
+	. "github.com/onsi/gomega"
+)
+
+var (
+	driverResourcePath        = "https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/master/deploy"
+	providerResourceDirectory = "manifest_staging/deployment"
+
+	driverResources = []string{
+		"csidriver.yaml",
+		"rbac-secretproviderclass.yaml",
+		"rbac-secretproviderrotation.yaml",
+		"rbac-secretprovidersyncing.yaml",
+		"secrets-store-csi-driver-windows.yaml",
+		"secrets-store-csi-driver.yaml",
+		"secrets-store.csi.x-k8s.io_secretproviderclasses.yaml",
+		"secrets-store.csi.x-k8s.io_secretproviderclasspodstatuses.yaml",
+	}
+
+	providerResources = []string{
+		"provider-azure-installer.yaml",
+	}
+)
+
+// InstallManifest install driver and provider manifests from yaml files.
+func InstallManifest(kubeconfigPath string, config *framework.Config) {
+	for _, resource := range driverResources {
+		err := exec.KubectlApply(kubeconfigPath, framework.NamespaceKubeSystem, []string{"-f", fmt.Sprintf("%s/%s", driverResourcePath, resource)})
+		Expect(err).To(BeNil())
+	}
+
+	wd, err := os.Getwd()
+	Expect(err).To(BeNil())
+
+	providerResourceAbsolutePath, err := filepath.Abs(fmt.Sprintf("%s/../../%s", wd, providerResourceDirectory))
+	Expect(err).To(BeNil())
+
+	for _, resource := range providerResources {
+		file, err := os.Open(fmt.Sprintf("%s/%s", providerResourceAbsolutePath, resource))
+		Expect(err).To(BeNil())
+
+		fileBytes, err := ioutil.ReadAll(file)
+		Expect(err).To(BeNil())
+
+		// Extract DS yaml
```

Comment:
Could you more info in the comment on why we need to extract the daemonset yaml from the single yaml?

Reply:
updated comment